Tests/Resources/app/AppKernel.php

```diff
@@ -86,4 +87,32 @@ public function indexAction()
     {
         return new Response();
     }
+
+    protected function build(ContainerBuilder $container)
+    {
+        $container->addCompilerPass(new PublicServicesForFunctionalTestsPass());
+    }
+}
+
+class PublicServicesForFunctionalTestsPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        $services = [
+            'httplug.strategy',
+            'httplug.auto_discovery.auto_discovered_client',
+            'httplug.auto_discovery.auto_discovered_async',
+            'httplug.message_factory.default',
+            'httplug.stream_factory.default',
+            'httplug.uri_factory.default',
+            'httplug.async_client.default',
+            'httplug.client.default',
+            'app.http.plugin.custom',
+        ];
+        foreach ($services as $service) {
+            if ($container->hasDefinition($service)) {
+                $container->getDefinition($service)->setPublic(true);
```

Comment:
this could be quite confusing when reading the tests, or when creating a new test. is there any chance we can instead have our tests operate on the container builder instead of the closed container?

Reply:
Yes, that's the idea but it's too many work ATM.